### ISSUE TITLE
Alter Database Unparsable Section

### ISSUE THREAD

**Issue by mariugul** [sqlfluff/sqlfluff#2017]
## Expected Behaviour
Using the postgres [ALTER DATABASE](https://www.postgresql.org/docs/14/sql-alterdatabase.html) should work.

## Observed Behaviour
```bash
$ sqlfluff lint --dialect postgres alter.sql
== [alter.sql] FAIL                                                          
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER
                       | DATABASE postgres SET password_enc...'
All Finished 📜 🎉!
```

## Steps to Reproduce

**alter.sql**:

```sql
ALTER DATABASE postgres SET password_encryption TO 'scram-sha-256';
```

```bash
sqlfluff lint --dialect postgres alter.sql
```

## Dialect
`postgres`

## Version
`sqlfluff, version 0.8.2`
`Python 3.8.10`
`pip 20.0.2`

## Configuration
`--dialtect postgres`
